lib/trento/software_updates/settings.ex

```diff
@@ -43,14 +47,47 @@ defmodule Trento.SoftwareUpdates.Settings do
     end
   end
 
-  defp maybe_validate_ca_cert(changeset, settings_submission) do
-    if nil != Map.get(settings_submission, :ca_cert) do
-      validate_required(changeset, :ca_cert)
+  defp validate_ca_cert(_ca_cert_atom, ca_cert) do
+    with {:ok, certificate} <- parse_ca_cert(ca_cert),
+         :ok <- ca_cert_valid?(certificate) do
+      []
     else
-      changeset
+      {:error, errors} -> errors
+    end
+  end
+
+  defp parse_ca_cert(ca_cert) do
+    case X509.Certificate.from_pem(ca_cert) do
+      {:ok, _} = result ->
+        result
+
+      _ ->
+        {:error, [ca_cert: {"unable to parse X.509 certificate", validation: :ca_cert_parsing}]}
+    end
+  end
+
+  defp ca_cert_valid?(certificate) do
+    now = DateTime.utc_now()
+    {:Validity, never_before, never_after} = X509.Certificate.validity(certificate)
+
+    if never_before |> X509.DateTime.to_datetime() |> DateTime.before?(now) and
+         never_after |> X509.DateTime.to_datetime() |> DateTime.after?(now) do
+      :ok
+    else
+      {:error, [ca_cert: {"the X.509 certificate is not valid", validation: :ca_cert_validity}]}
     end
   end
 
+  defp maybe_validate_ca_cert(changeset, %{ca_cert: nil}), do: changeset
```

Comment:
Nothing major. Would you mind moving these `maybe_validate_ca_cert` before `validate_ca_cert`, please?
Reading would just flow smoother :smile: